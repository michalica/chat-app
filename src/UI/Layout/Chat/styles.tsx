import bgImg from "../../../assets/img/BodyBG.png";


const styles = {
    messageArea: {
        backgroundImage: `url(${bgImg})`,
        height: '90vh',
    },
    userArea: {
        height: '10vh',
        background: '#3898D4'
    },
};

export default styles;
