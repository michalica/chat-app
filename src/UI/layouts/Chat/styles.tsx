import bgImg from "../../../assets/img/BodyBG.png";

const styles = {
  messageArea: {
    backgroundImage: `url(${bgImg})`,
    height: "calc(100vh - 80px)",
    overflowY: "scroll" as "scroll",
  },
  userArea: {
    height: 80,
    background: "#3898D4",
    position: "fixed" as "fixed",
    bottom: 0,
    left: 0,
    right: 0,
  },
};

export default styles;
