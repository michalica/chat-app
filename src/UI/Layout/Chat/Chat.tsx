import React, {ReactNode} from 'react';
import {makeStyles} from "@material-ui/core";
import styles from "./styles";

const useStyles = makeStyles(styles);

const Chat = (props: ChatProps) => {

    const {
        messageArea,
        userArea,
    } = props;

    const classes = useStyles();
    return (
        <div>
            <div
                className={classes.messageArea}
            >
                {messageArea}
            </div>
            <div
                className={classes.userArea}
            >
                {userArea}
            </div>
        </div>
    );
};

export interface ChatProps
{
    messageArea: ReactNode;
    userArea: ReactNode;
}

export default Chat;
