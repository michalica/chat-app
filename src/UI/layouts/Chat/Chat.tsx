import React, {ReactNode} from 'react';
import {makeStyles} from "@material-ui/core";
import styles from "./styles";
import Container from '../../atoms/Container';

const useStyles = makeStyles(styles);

const Chat = (props: ChatProps) => {

    const {
        messageArea,
        userArea,
    } = props;

    const classes = useStyles();
    return (
        <div>
            <div
                className={classes.messageArea}
            >
                <Container>
                    {messageArea}
                </Container>
            </div>
            <div
                className={classes.userArea}
            >
                <Container>
                    {userArea}
                </Container>
            </div>
        </div>
    );
};

export interface ChatProps
{
    messageArea: ReactNode;
    userArea: ReactNode;
}

export default Chat;


