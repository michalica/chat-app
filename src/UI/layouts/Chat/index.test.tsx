import Chat from "./Chat";
import { shallow} from "enzyme";

describe("snapshot testing", () => {
    it('should render view correctly', () => {
        expect(
            shallow(<Chat
            messageArea={<div>message area</div>}
            userArea={<div>user area</div>}
        />)
        ).toMatchSnapshot();
    });
})
