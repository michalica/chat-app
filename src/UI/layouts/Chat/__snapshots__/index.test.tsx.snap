// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should render view correctly 1`] = `
<div>
  <div
    className="makeStyles-messageArea-1"
  >
    <div>
      message area
    </div>
  </div>
  <div
    className="makeStyles-userArea-2"
  >
    <div>
      user area
    </div>
  </div>
</div>
`;
